ledger/src/blockstore_processor.rs

```diff
@@ -1569,8 +1593,50 @@ pub mod tests {
         trees::tr,
     };
 
+    // Convenience wrapper to optionally process blockstore with Secondary access.
+    //
+    // Setting up the ledger for a test requires Primary access as items will need to be inserted.
+    // However, once a Secondary access has been opened, it won't automaticaly see updates made by
+    // the Primary access. So, open (and close) the Secondary access within this function to ensure
+    // that "stale" Secondary accesses don't propagate.
+    fn test_process_blockstore_with_custom_options(
+        genesis_config: &GenesisConfig,
+        blockstore: &Blockstore,
+        opts: ProcessOptions,
```

Comment:
Looks like we can simply pass the reference here to avoid cloning on the call-site as `test_process_blockstore` also uses the reference of `opts` internally.

https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore_processor.rs#L575-L583

Reply:
Definitely going to resolve this one separately, so as not to muddy up this diff.